internal/provider/network/routed_resource.go

```diff
@@ -0,0 +1,540 @@
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving VDC", err.Error())
+		return
+	}
+
+	parentEdgeGatewayOwnerID, errGet := getParentEdgeGatewayID(org, state.EdgeGatewayID.ValueString())
+	if errGet != nil {
+		resp.Diagnostics.AddError(errGet.Summary, errGet.Detail)
+		return
+	}
+
+	if govcd.OwnerIsVdcGroup(*parentEdgeGatewayOwnerID) {
+		networkMutexKV.KvLock(ctx, *parentEdgeGatewayOwnerID)
+		defer networkMutexKV.KvUnlock(ctx, *parentEdgeGatewayOwnerID)
+	} else {
+		networkMutexKV.KvLock(ctx, state.EdgeGatewayID.ValueString())
+		defer networkMutexKV.KvUnlock(ctx, state.EdgeGatewayID.ValueString())
+	}
+
+	orgNetwork, err := org.GetOpenApiOrgVdcNetworkById(state.ID.ValueString())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			tflog.Debug(ctx, "Network not found, removing resource from state")
+			resp.State.RemoveResource(ctx)
+		}
+		resp.Diagnostics.AddError("Error retrieving routing network", err.Error())
+		return
+	}
+	err = orgNetwork.Delete()
+	if err != nil {
+		resp.Diagnostics.AddError("Error deleting routing network", err.Error())
+		return
+	}
+}
+
+func (r *networkRoutedResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resourceURI := strings.Split(req.ID, ".")
+
+	if len(resourceURI) != 2 {
+		resp.Diagnostics.AddError("Error importing network_routed", "Resource name must be specified as vdc-name.network-name or vdc-group-name.network-name")
+		return
+	}
+
+	vdcOrVDCGroupName, networkName := resourceURI[0], resourceURI[1]
+
+	org, vdc, err := r.client.GetOrgAndVDC(r.client.GetOrg(), r.client.GetDefaultVDC())
+	if err != nil && govcd.ContainsNotFound(err) {
+		resp.Diagnostics.AddError("Error retrieving VDC", err.Error())
+		return
+	}
+
+	orgNetwork, err := vdc.GetOpenApiOrgVdcNetworkByName(networkName)
+	if err != nil && !govcd.ContainsNotFound(err) {
+		resp.Diagnostics.AddError("Error retrieving org vdc network by name", err.Error())
+		return
+	}
+
+	if orgNetwork == nil {
+		adminOrg, err := r.client.Vmware.GetAdminOrgByName(org.Org.Name)
+		if err != nil {
+			resp.Diagnostics.AddError("Error retrieving Admin Org", err.Error())
+			return
+		}
+		vdcgroup, err := adminOrg.GetVdcGroupByName(vdcOrVDCGroupName)
+		if err != nil {
+			resp.Diagnostics.AddError("Error retrieving VDC group", err.Error())
+			return
+		}
+		orgNetwork, err = vdcgroup.GetOpenApiOrgVdcNetworkByName(networkName)
+		if err != nil && govcd.ContainsNotFound(err) {
+			resp.Diagnostics.AddError("Error retrieving org vdc_group network by name", err.Error())
+			return
+		}
+	}
+
+	if !orgNetwork.IsRouted() {
+		resp.Diagnostics.AddError("Error importing routed network", fmt.Sprintf("Org network with name '%s' found, but is not of type Routed (type is '%s')", networkName, orgNetwork.GetType()))
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("name"), networkName)...)
```

Comment:
Why do you set attribute `name`? this attribute seems to not be used in Read function.

Reply:
Removed